### PR TITLE
feat(core): init editor setting ui

### PR DESCRIPTION
**Pull request by JimmFly**
- init editor setting ui

https://github.com/user-attachments/assets/c54f5816-ef05-4ac0-b11a-8ab7159f928c

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#7924** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7924?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7878** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7878?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7879** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7879?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/component/src/index.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ export * from './ui/popover';
 export * from './ui/radio';
 export * from './ui/scrollbar';
 export * from './ui/skeleton';
+export * from './ui/slider';
```
Comment (pengx17) [inline packages/frontend/component/src/index.ts]: I think the slider changes should go into another PR

File: packages/frontend/component/src/ui/slider/slider.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+export interface SliderProps {
+  value?: number;
+  onChange?: (value: number) => void;
+  min?: number;
+  max?: number;
+  step?: number;
+  nodes?: number[]; // The values where the nodes should be placed
+}
+
+export const Slider = ({
```
Comment (pengx17) [inline packages/frontend/component/src/ui/slider/slider.tsx]: please impl based on https://www.radix-ui.com/primitives/docs/components/slider for better keyboard + a11y support

Comment (JimmFly) [reply packages/frontend/component/src/ui/slider/slider.tsx]: I have tried using Radix-UI, but the thumb does not move proportionally, causing the stepNode and thumb to be misaligned. This issue is currently blocking me from using Radix-UI.

File: packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/edgeless/shape.tsx [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+  return (
+    <>
+      <EdgelessSnapshot
+        title={t['com.affine.settings.editorSettings.edgeless.shape']()}
+        option={['mock-option']}
+        type="mock-type"
+      />
+      <div>
+        <Button variant="plain">
```
Comment (CatsJuice) [inline packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/edgeless/shape.tsx]: maybe better to use `<RadioGroup />`, track #7850

File: packages/common/env/src/global.ts [hunk 4]
```diff
@@ -32,6 +32,7 @@ export const runtimeFlagsSchema = z.object({
   enableInfoModal: z.boolean(),
   enableOrganize: z.boolean(),
   enableThemeEditor: z.boolean(),
+  enableEditorSettings: z.boolean(),
```
Comment (EYHN) [inline packages/common/env/src/global.ts]: move to https://github.com/toeverything/AFFiNE/blob/e01ce2cd9163dfe751c5e8979bfe5a632e0a9f3d/packages/common/infra/src/modules/feature-flag/constant.ts#L8

**Review by EYHN — APPROVED** [review]
